TIL.md

```diff
@@ -0,0 +1,249 @@
+#Thursday
+
+##Vim
+  $ - end of the line
+  0 - beginning of the line
+  c - change
+  d - delete 
+  DD - remove whole line
+  P - paste 
+  v - visual mode
+  i - insert mode
+  r - replace
+  Esc - returns to normal mode
+
+## Page object collection
+- collection option of the page objects allows to greatly simplify code, product-thumb can be found 
+```` javascript
+  products: collection({
+    itemScope: '[data-test-product-thumb]',
+    item: {
+      open: clickable('[data-test-product-thumb-link]'),
+      delete: clickable('[data-test-product-destroy-button]'),
+      cancelDelete: clickable('[data-test-product-destroy-cancel-button]'),
+      confirmDelete: clickable('[data-test-product-destroy-confirm-button]')
+    }
+  })
+````
+- functions on page objects can be chained
+```` javascript
+  await page.openProducts()
+    .openNewProduct()
+    .editProductName('FNX 45')
+    .saveProduct();
+````
+- sinon.js 
+
+#Wednesday
+
+# Tuesday
+
+## ember-cli-sass
+
+## Useful testing packages
+https://github.com/cibernox/ember-native-dom-helpers
+https://github.com/simplabs/ember-test-selectors
+http://ember-cli-page-object.js.org/docs/v1.8.x/
+
+## Components
+- new components were added
+  - edit which is a form because of convention and proper setting of route in route.js, 
+    this.route('products', function() {
+    this.route('edit', { path: '/:product_id/edit' });,
+    explicit search for the product was not necessary (:product_id is very important in this)
+- actions like save() are declared like that, in component.js
+  ```` js
+  export default Component.extend({
+    actions: {
+      save() {
+        this.product.save()
+          .catch((error) => alert(error));
+      }
+    }
+  });
+  ````
+(edit, thumb, actions:[])
+
+## Integration component test
+- integration tests ensure that component renders properly with given data, this cannot test transition between pages because there is no router in place 
+(we ensure component renders properly with given data, (no transition testing, link-to) the process of transition from .. to ..)
+  ```` js
+  setupComponentTest('product-edit', {
+    integration: true
+  });
+
+  it('renders', function() {
+    this.set('product', { name: 'Phone', description: 'Black', sku: 1 });
+    this.render(hbs`{{product-edit product=product}}`);
+    expect(find('[data-test-product-form-name]').value).to.include('Phone');
+  });
+````
+## Acceptance test
```

Comment:
Improve formatting of the document, it's pretty hard to read it. With more cautions about headers, insert empty lines as spacer where appropriate, ensure that code samples are properly formatted

https://github.com/MattSkrobis/bsc-ember/blob/176fb728caa69139404314761f132018089dc11d/TIL.md